pkgs/unified_analytics/lib/src/constants.dart

```diff
@@ -2,6 +2,9 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
+/// The hard coded client ID for the [NoOpAnalytics] instance.
+const String clientIdNoOp = 'xxxx-xxxx';
```

Comment:
I think this would make more sense as a static member on the `NoOpAnalytics` class. But also, it's not clear to me that having a reference to this separate from calling the `clientId` getter is any better, as you're just testing the implementation of `NoOpAnalytics`.

Reply:
I could see it being useful as a static method, I added this originally as a constant so that I can use it in tests that are looking to confirm that instance is a NoOp instance. This would be better than just hard coding the NoOp id inside the tests in flutter tools

Reply:
> I added this originally as a constant so that I can use it in tests that are looking to confirm that instance is a NoOp instance.

You could assert that the runtime type is `NoOpAnalytics`.